Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -149,10 +155,13 @@ void OpenALStream::SoundLoop()
   memset(uiBuffers, 0, numBuffers * sizeof(ALuint));
   uiSource = 0;
 
-  // Checks if a X-Fi is being used. If it is, disable FLOAT32 support as this sound card has no
-  // support for it even though it reports it does.
+  if (alIsExtensionPresent("AL_EXT_float32"))
+    float32_capable = true;
+
+  // Checks if a X-Fi is being used. If it is, enable FIXED32 support as this sound card
+  // is the only one that supports it for now.
   if (strstr(alGetString(AL_RENDERER), "X-Fi"))
-    float32_capable = false;
+    fixed32_capable = true;
```

Comment:
Can you extend the comment to explain why you (have to) check for the name instead of a registered extension or anything like that?